runtime/tests/checker/interface_test.go

```diff
@@ -2689,3 +2689,1459 @@ func TestCheckBadStructInterface(t *testing.T) {
+
+		errs := RequireCheckerErrors(t, err, 2)
+
+		conformanceError := &sema.ConformanceError{}
+		require.ErrorAs(t, errs[0], &conformanceError)
+		assert.Equal(t, "B", conformanceError.InterfaceType.QualifiedIdentifier())
+		assert.Equal(t, "A", conformanceError.NestedInterfaceType.QualifiedIdentifier())
+
+		require.ErrorAs(t, errs[1], &conformanceError)
+		assert.Equal(t, "Q", conformanceError.InterfaceType.QualifiedIdentifier())
+		assert.Equal(t, "P", conformanceError.NestedInterfaceType.QualifiedIdentifier())
+	})
+
+	t.Run("same conformance via different paths", func(t *testing.T) {
+
+		t.Parallel()
+
+		_, err := ParseAndCheck(t, `
+            struct interface A {
+                pub fun hello() {
+                    var a = 1
+                }
+            }
+
+            struct interface P: A {}
+
+            struct interface Q: A {}
+
+            struct interface X: P, Q {}
+        `)
+
+		require.NoError(t, err)
+	})
+
+	t.Run("same conformance via different paths for struct", func(t *testing.T) {
+
+		t.Parallel()
+
+		_, err := ParseAndCheck(t, `
+            struct interface A {
+                pub fun hello() {
+                    var a = 1
+                }
+            }
+
+            struct interface P: A {}
+
+            struct interface Q: A {}
+
+            struct X: P, Q {}
+        `)
+
+		require.NoError(t, err)
+	})
+
+	t.Run("cyclic conformance", func(t *testing.T) {
+
+		t.Parallel()
+
+		_, err := ParseAndCheck(t, `
+            struct interface Foo: Baz {
+                let x: Int
+
+                fun test(): Int
+            }
+
+            struct interface Bar: Foo {}
+
+            struct interface Baz: Bar {}
+        `)
+
+		errs := RequireCheckerErrors(t, err, 3)
+
+		conformanceError := sema.CyclicConformanceError{}
+		require.ErrorAs(t, errs[0], &conformanceError)
+		assert.Equal(t, "Foo", conformanceError.InterfaceType.QualifiedIdentifier())
+
+		require.ErrorAs(t, errs[1], &conformanceError)
+		assert.Equal(t, "Bar", conformanceError.InterfaceType.QualifiedIdentifier())
+
+		require.ErrorAs(t, errs[2], &conformanceError)
+		assert.Equal(t, "Baz", conformanceError.InterfaceType.QualifiedIdentifier())
+	})
+}
+
+func TestCheckInterfaceDefaultMethodsInheritance(t *testing.T) {
+
+	t.Parallel()
+
+	t.Run("default impl in super", func(t *testing.T) {
+
+		t.Parallel()
+
+		_, err := ParseAndCheck(t, `
+            struct interface A {
+                pub fun hello() {
+                    var a = 1
+                }
+            }
+
+            struct interface B: A {
+                pub fun hello()
+            }
+        `)
+
+		require.NoError(t, err)
+	})
+
+	t.Run("default impl in child", func(t *testing.T) {
+
+		t.Parallel()
+
+		_, err := ParseAndCheck(t, `
+            struct interface A {
+                pub fun hello()
+            }
+
+            struct interface B: A {
+                pub fun hello() {
+                    var a = 1
+                }
+            }
```

Comment:
Currently, it is an error when a concrete type conforms to interfaces where there is a requirement and a default function (#2471), e.g. having just `struct interface B` and adding `struct C: A, B` here.

Should this also be enforced in the interface inheritance case?

Reply:
I'll also add the same restriction here, and probably document this as unimplemented/deviation from the FLIP. 👍

Reply:
I see, I had not realized that the FLIP specified a different behaviour. Though maybe different than from the FLIP, it might be better to be consistent and avoid confusion for developers.

Like discussed and suggested by you, we can then maybe later put out a FLIP and propose to lift the restrictions everywhere, keeping a consistent behaviour.

Reply:
Added the restriction here: ed64d6f2a611ea5280862a2bf6fb85e4f84afdcd